storage/tests/unit/test_bucket.py

```diff
@@ -569,6 +569,21 @@ def test_get_blob_hit_w_user_project(self):
         self.assertEqual(kw["path"], "/b/%s/o/%s" % (NAME, BLOB_NAME))
         self.assertEqual(kw["query_params"], {"userProject": USER_PROJECT})
 
+    def test_get_blob_hit_w_generation(self):
+        NAME = "name"
+        BLOB_NAME = "blob-name"
+        GENERATION = 1512565576797178
+        connection = _Connection({"name": BLOB_NAME})
+        client = _Client(connection)
+        bucket = self._make_one(name=NAME)
+        blob = bucket.get_blob(BLOB_NAME, client=client, generation=GENERATION)
```

Comment:
This might be overdoing it a bit, but can you add a unit test for bucket.blob() as well? Additionally could you add a test for the constructor of the Blob class?